adapters/src/test/kafka.rs

```diff
@@ -0,0 +1,261 @@
+use crate::test::{wait, TestStruct};
+use csv::{ReaderBuilder as CsvReaderBuilder, WriterBuilder as CsvWriterBuilder};
+use futures::executor::block_on;
+use rdkafka::{
+    admin::{AdminClient, AdminOptions, NewPartitions, NewTopic, TopicReplication},
+    client::DefaultClientContext,
+    config::{FromClientConfig, RDKafkaLogLevel},
+    consumer::{BaseConsumer, Consumer},
+    producer::{BaseRecord, DefaultProducerContext, Producer, ThreadedProducer},
+    util::Timeout,
+    ClientConfig, Message,
+};
+use std::{
+    sync::{
+        atomic::{AtomicBool, Ordering},
+        Arc, Mutex,
+    },
+    thread,
+    thread::JoinHandle,
+    time::{Duration, SystemTime, UNIX_EPOCH},
+};
+
+pub struct KafkaResources {
+    admin_client: AdminClient<DefaultClientContext>,
+    topics: Vec<String>,
+}
+
+/// An object that creates a couple of topics on startup and deletes them
+/// on drop.  Helps make sure that test runs don't leave garbage behind.
+impl KafkaResources {
+    pub fn create_topics(topics: &[(&str, i32)]) -> Self {
```

Comment:
Most of this code is async, this should probably be an async function instead of internally blocking

Reply:
I use the sync version of the Kafka API.  There are a couple of APIs that don't have a sync version (presumably an omission), so I wrap them in `block_on`, but I don't want to propagate asynchronicity to the rest of the framework.